docs/user-guides/scripts.md

```diff
@@ -0,0 +1,423 @@
+
+#### `None` loader
+With `None` set as the loader (which is by default) in the Artifact annotation, the `path` attribute of `Artifact` is
+extracted and used to provide a `pathlib.Path` object for the given argument, which can be used directly in the function
+body. The following example is the same as above except for explicitly setting the loader to `None`:
+
+```python
+@script(constructor="runner")
+def read_artifact(
+    an_artifact: Annotated[Path, Artifact(name="my-artifact", path="/tmp/file", loader=None)]
+):
+    print(an_artifact.read_text())
+```
+
+#### `file` loader
+
+When the loader is set to `file`, the function parameter type should be `str`, and will contain the contents string
+representation of the file stored at `path` (essentially performing `path.read_text()` automatically):
+
+```python
+@script(constructor="runner")
+def read_artifact(
+    an_artifact: Annotated[str, Artifact(name="my-artifact", path="/tmp/file", loader=ArtifactLoader.file)]
+) -> str:
+    return an_artifact
+```
+
+This loads the contents of the file at `"/tmp/file"` to the argument `an_artifact` and subsequently can be used as a
+string inside the function.
+
+#### `json` loader
+
+When the loader is set to `json`, the contents of the file at `path` are read and parsed to a dictionary via `json.load`
+(essentially performing `json.load(path.open())` automatically). By specifying a Pydantic type, this dictionary can even
+be automatically parsed to that type:
+
+```python
+class MyArtifact(BaseModel):
+    a = "a"
+    b = "b"
+
+
+@script(constructor="runner")
+def read_artifact(
+    an_artifact: Annotated[MyArtifact, Artifact(name="my-artifact", path="/tmp/file", loader=ArtifactLoader.json)]
+) -> str:
+    return an_artifact.a + an_artifact.b
+```
+
+Here, we have a json representation of `MyArtifact` such as `{"a": "hello ", "b": "world"}` stored at `"/tmp/file"`. We
+can load it with `ArtifactLoader.json` and then use `an_artifact` as an instance of `MyArtifact` inside the function, so
+the function will return `"hello world"`.
+
+#### Function parameter name aliasing
+
+Script annotations can work on top of the `RunnerScriptConstructor` for name aliasing of function
+parameters, in particular to allow a public `kebab-case` parameter, while using a `snake_case`
+Python function parameter. When using a `RunnerScriptConstructor`, an environment variable
+`hera__script_annotations` will be added to the Script template (visible in the exported YAML file).
+
+### Outputs
+
+> Note: Output annotations are only supported when used with the `RunnerScriptConstructor`.
+
+Annotations can also be used for outputs. An example can be seen
+[here](../../examples/workflows/script_annotations_outputs).
+
+For a simple hello world output artifact example we currently have:
+```python
+@script(outputs=Artifact(name="hello-art", path="/tmp/hello_world.txt"))
+def whalesay():
```

Comment:
We should probably make this function and the below one the same
```python
@script(outputs=Artifact(name="hello-art", path="/tmp/hello_world.txt"))
def hello_world():
    with open("/tmp/hello_world.txt", "w") as f:
        f.write("Hello, World!")
```
and 
```python
@script()
def hello_world() -> Annotated[str, Artifact(name="hello-art", path="/tmp/hello_world.txt")]:
    return "Hello, World!"
```

Reply:
Lol didn't realise they had weird naming 😆